core/src/verified_vote_packets.rs

```diff
@@ -158,12 +209,39 @@ impl VerifiedVotePackets {
                     let slot = vote.last_voted_slot().unwrap();
                     let hash = vote.hash();
 
-                    let validator_votes = self.0.entry(vote_account_key).or_default();
-                    validator_votes.insert((slot, hash), (packet_batch, signature));
-
-                    if validator_votes.len() > MAX_VOTES_PER_VALIDATOR {
-                        let smallest_key = validator_votes.keys().next().cloned().unwrap();
-                        validator_votes.remove(&smallest_key).unwrap();
+                    if new_style {
+                        let latest_slot_sent_to_bank = match self.0.get(&vote_account_key) {
+                            Some(&FullVote((slot, _), _)) => slot,
+                            _ => 0,
+                        };
+                        // Since votes are not incremental, we keep only the latest vote
+                        if slot > latest_slot_sent_to_bank {
+                            self.0.insert(
+                                vote_account_key,
+                                FullVote((slot, hash), (packet_batch, signature)),
```

Comment:
If somebody tries to submit an incremental vote, that will also overwrite their latest full vote right? This means people submitting incremental votes will only have their latest vote show up as well, potentially meaning their towers are more out of sync.

This might be the right thing to do to prevent people from trying to get more rewards by submitting the incremental votes over the full votes. But if we're going to shut down support for incremental votes here, maybe we should start thinking about doing that across the board.

Reply:
Good point, I changed it so that we only store a full vote if the feature is active and they have sent a VoteStateUpdate. This shouldn't change any behavior for those sending incremental votes now.